internal/logging/context.go

```diff
@@ -15,39 +17,113 @@ import (
 
 type loggerContextKey struct{}
 
-var globalLogger *log.Entry
+var (
+	globalLogger logr.Logger
+
+	// programLevel allows for setting the logging level dynamically
+	// via SetLevel().
+	programLevel = new(slog.LevelVar)
+)
+
+const (
+	LevelTrace = slog.Level(-2)
+	LevelDebug = slog.Level(-1)
+	LevelInfo  = slog.Level(0)
+	LevelError = slog.Level(8)
+
+	// These constants contain logging level strings,
+	// purely for the performance benefit.
+	TRACE = "TRACE"
+	DEBUG = "DEBUG"
+	INFO  = "INFO"
+	ERROR = "ERROR"
+)
 
 func init() {
-	globalLogger = log.New().WithFields(nil)
-	level, err := log.ParseLevel(os.GetEnv("LOG_LEVEL", "INFO"))
+	level, err := parseLevel(os.GetEnv("LOG_LEVEL", "INFO"))
 	if err != nil {
 		panic(err)
 	}
-	globalLogger.Logger.SetLevel(level)
+	programLevel.Set(level)
+
+	opts := &slog.HandlerOptions{
+		Level: programLevel,
+		ReplaceAttr: func(groups []string, a slog.Attr) slog.Attr {
+			// Customize the name of the level key and the output string,
+			// including custom level values. Meaning:
+			// log.V(1).Info() becomes level=DEBUG instead of level=DEBUG+3
+			// log.V(2).Info() becomes level=TRACE instead of level=DEBUG+2
+			if a.Key == slog.LevelKey {
+				// Handle custom level values.
+				level := a.Value.Any().(slog.Level)
+
+				switch {
+				case level < LevelDebug:
+					a.Value = slog.StringValue(TRACE)
+				case level < LevelInfo:
+					a.Value = slog.StringValue(DEBUG)
+				case level < LevelError:
+					a.Value = slog.StringValue(INFO)
+				default:
+					a.Value = slog.StringValue(ERROR)
+				}
+			}
+			return a
+		},
+	}
+
+	globalLogger = logr.FromSlogHandler(slog.NewTextHandler(os.Stderr, opts))
+
 	SetKLogLevel(os.GetEnvInt("KLOG_LEVEL", 0))
```

Comment:
I wonder how much of https://github.com/go-logr/logr#using-a-logrlogsink-as-backend-for-slog is applicable here...